loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/LoadFlow.java

```diff
@@ -6,15 +6,119 @@
+                .filter(p -> p.getName().equals(name))
+                .findFirst()
+                .orElseThrow(() -> new PowsyblException("Loadflow '" + name + "' not found"));
+
+        return new LoadFlow(provider, platformConfig);
+    }
+
+    public static LoadFlow findDefault() {
+        return findDefault(PROVIDERS_SUPPLIERS.get(), PlatformConfig.defaultConfig());
+    }
+
+    static LoadFlow findDefault(List<LoadFlowProvider> providers, PlatformConfig platformConfig) {
+        Objects.requireNonNull(providers);
+        Objects.requireNonNull(platformConfig);
+
+        if (providers.isEmpty()) {
+            throw new PowsyblException("No loadflow providers found");
+        }
+        if (providers.size() == 1) {
+            return new LoadFlow(providers.get(0), platformConfig);
+        } else {
+            // try to find a loadflow config to known which implementation has to be chosen
+            String name = platformConfig.getOptionalModuleConfig("load-flow")
+                    .flatMap(mc -> mc.getOptionalStringProperty("default"))
+                    .orElseThrow(() -> new PowsyblException("Loadflow configuration not found"));
+            return find(name, providers, platformConfig);
+        }
+    }
+
+    public CompletableFuture<LoadFlowResult> runAsync(Network network, String workingStateId, ComputationManager computationManager, LoadFlowParameters parameters) {
+        Objects.requireNonNull(workingStateId);
+        Objects.requireNonNull(parameters);
+        return provider.run(network, computationManager, workingStateId, parameters);
+    }
+
+    public CompletableFuture<LoadFlowResult> runAsync(Network network, ComputationManager computationManager, LoadFlowParameters parameters) {
+        return runAsync(network, network.getVariantManager().getWorkingVariantId(), computationManager, parameters);
+    }
+
+    public CompletableFuture<LoadFlowResult> runAsync(Network network, LoadFlowParameters parameters) {
+        return runAsync(network, LocalComputationManager.getDefault(), parameters);
+    }
+
+    public CompletableFuture<LoadFlowResult> runAsync(Network network) {
+        return runAsync(network, LoadFlowParameters.load(platformConfig));
+    }
+
+    public LoadFlowResult run(Network network, String workingStateId, ComputationManager computationManager, LoadFlowParameters parameters) {
+        Objects.requireNonNull(workingStateId);
+        Objects.requireNonNull(parameters);
+        return provider.run(network, computationManager, workingStateId, parameters).join();
+    }
+
+    public LoadFlowResult run(Network network, ComputationManager computationManager, LoadFlowParameters parameters) {
+        return run(network, network.getVariantManager().getWorkingVariantId(), computationManager, parameters);
+    }
+
+    public LoadFlowResult run(Network network, LoadFlowParameters parameters) {
+        return run(network, LocalComputationManager.getDefault(), parameters);
```

Comment:
`LocalComputationManager.getDefault()` depends on `PlatformConfig.defaultConfig()`. Shouldn't it use the local `platformConfig` attribute?

Reply:
But there is no way to do that because PlatformConfig has to be pass to LocalComputationManager constructor, and that is not possible here with LocalComputartionManager.getDefault().